app/assets/javascripts/galleries/uploader.js

```diff
@@ -40,6 +42,17 @@ function bindFileInput(fileInput, form, submitButton, formData) {
     imageMaxHeight: 400,
 
     add: function(e, data) {
+      if (typeof limit !== 'undefined' && limit > 1) {
+        var isFirstFile = (fileInput[0].files[0] == data.files[0]);
+        var numUploading = fileInput[0].files.length;
+        if (isFirstFile) numFiles += numUploading;
+        if (numFiles > limit) {
+          if (isFirstFile) alert("You cannot upload more than "+limit+" files at once. Please try again.");
+          if(fileInput[0].files[numUploading - 1] == data.files[0]) numFiles -= numUploading;
```

Comment:
...but once done complaining, we should recognize that they did not in fact upload any of these files and we should discount that fact. So if, say, limit is 2 and they've uploaded 1 and try 2 more, we should stop them but still allow them to upload 1 more.